### ISSUE TITLE
Release plan for 1.14.8

### ISSUE THREAD

**Issue by patricklodder** [dogecoin/dogecoin#3587]
There are several pending bug fixes on the 1.15.0-dev branch that benefit the network when they become available asap. Therefore, a 1.14.8 release will be beneficial. 

In scope: bug fixes, tests and build system fixes.
Out of scope: everything else.

Work to be backported from 1.15.0-dev:

- [x] Update the CI: #3584
    - #3467
    - #3517
    - #3582
- [x] Test fixes: #3586
    - #3572
    - #3574
    - #3446
- [x] TxRequestTracker fix (separate, because backport depends on other PRs): #3592 
    - #3577
- [x] Other bugfixes: #3591 
    - #3567
    - #3568
    - #3575
    - #3576
- [x] Pinned dependencies updates: #3590 
    - #3362
    - #3481
- [x] Enable build on systems with newer Boost versions up to 1.85: #3588
    - #3313
    - #3558
    - #3456
- [x] Enable build on systems with newer protobuf (this requires c++17): #3589
    - #3561
    - #3494
    - #3560
    - #3490
    - #3492
    - #3458

Maintainer tasks:

- [x] Open 1.14.8: #3585
- [x] Update docs
    - #3612
    - #3611
- [x] Write release notes: #3609
- [x] Fixate v1.14.8 in source #3615
- [x] Tag v1.14.8
- [x] Gitian builds
- [x] Fast-fwd 1.14-maint
- [x] Upload binaries and sha256sum files

**Comment by chromatic** [comment]
@patricklodder do you want to do the honors of tagging?

**Comment by patricklodder** [comment]
Tag done. I'll update with my LXC gitian build results.

**Comment by chromatic** [comment]
Hm, I'm already getting different results:

```
64fc5477b102950e47345d3acdc16f7df9fa53275075dff3bc4e4ebe32d141bf  dogecoin-1.14.8-osx-unsigned.dmg
eee529cc15d70454c9eff036e98f992b78cb8d2bd3b1d007cce8a4f9db69e767  dogecoin-1.14.8-osx-unsigned.tar.gz
171f1b234669c7e6a38522f8d78960b6652e72e36c598a96e0952de27c29fdea  dogecoin-1.14.8-osx64.tar.gz
082a115927a974e13c9edd5c64a56a1c71b16289c3ed7590375af50439457cfc  src/dogecoin-1.14.8.tar.gz
```

Given that even the tarball is different, let me stop here and verify that I have a sensible command. I'm using:

```
./gitian-build.sh --docker --setup --build --sign '<my address>' --verify 1.14.8
```

**Comment by patricklodder** [comment]
```
082a115927a974e13c9edd5c64a56a1c71b16289c3ed7590375af50439457cfc  src/dogecoin-1
.14.8.tar.gz
```

source tarball is the same for me. I'll go do a docker run and see if the difference is with the debootstrap vs docker base image, or that there is something wrong in the code.

**Comment by patricklodder** [comment]
@chromatic unfortunately my docker macOS hashes are the same as my lxc hashes. 2 different boxes in different countries (though both in EU), different base OSs, one with LXC old script, one with docker and the script you use.

Would be good to get a third build from someone else.

Things to check on your end in the meantime:

1. start from a clean slate (i.e. empty new dir)
2. download the latest version of the script 

```
curl -L -O https://raw.githubusercontent.com/dogecoin/dogecoin/1.14.8-dev/contrib/gitian-build.sh
```

3. drop `--verify`

That's all I can think of

**Comment by chromatic** [comment]
I get the same results I had before--not matching yours--from a clean slate without `--verify`, so let's put this on hold until a third build. I'll let the Windows and Linux builds run just to see what happens.

```
64fc5477b102950e47345d3acdc16f7df9fa53275075dff3bc4e4ebe32d141bf  dogecoin-1.14.8-osx-unsigned.dmg
eee529cc15d70454c9eff036e98f992b78cb8d2bd3b1d007cce8a4f9db69e767  dogecoin-1.14.8-osx-unsigned.tar.gz
171f1b234669c7e6a38522f8d78960b6652e72e36c598a96e0952de27c29fdea  dogecoin-1.14.8-osx64.tar.gz
082a115927a974e13c9edd5c64a56a1c71b16289c3ed7590375af50439457cfc  src/dogecoin-1.14.8.tar.gz
54ee89a1e6113b199d5fbfdcde2308094688defaaa0a68ef891926b9aa2cda72  dogecoin-osx-1.14-res.yml
```

**Comment by slightlyskepticalpotat** [comment]
For what it's worth, I tried building yesterday with the same system and
steps I used for 1.14.7 a few months back. Got lots of weird issues on both
LXC and Docker which I thought was weird since the scripts haven't really
changed.

May give it another shot today on a different system.

**Comment by patricklodder** [comment]
I've found the issue overnight - it's caused by the version of `gitian-builder`. After updating on a fresh box to devrandom/gitian-builder@874ba2036e4bf97ac920257c36ff29e4dea78449, I get the same results as @chromatic for macOS.

I'm doing another run - consider my previous results void.

**Comment by patricklodder** [comment]
LXC shasums (this time with https://github.com/devrandom/gitian-builder/commit/874ba2036e4bf97ac920257c36ff29e4dea78449)

```
64fc5477b102950e47345d3acdc16f7df9fa53275075dff3bc4e4ebe32d141bf  dogecoin-1.14.8-osx-unsigned.dmg
eee529cc15d70454c9eff036e98f992b78cb8d2bd3b1d007cce8a4f9db69e767  dogecoin-1.14.8-osx-unsigned.tar.gz
171f1b234669c7e6a38522f8d78960b6652e72e36c598a96e0952de27c29fdea  dogecoin-1.14.8-osx64.tar.gz
```

**Comment by chromatic** [comment]
My results match Patrick's. One more contribution would be great!

**Comment by xanimo** [comment]
my source tarball is the same as both of yours but i've got different results :confused::
```
1f9682138e4207b89d62eba8134438dbcd437d7a8297cec7cd48d7355475bfed  dogecoin-1.14.8-aarch64-linux-gnu-debug.tar.gz
2c7d58f28637596dd48ebf3c45d46b5964cba47048f058a5183c65ebf2e9fba5  dogecoin-1.14.8-aarch64-linux-gnu.tar.gz
a11bdfb784baecc57753d9440875751dc224aee9a5f0f24faf49533b594542d2  dogecoin-1.14.8-arm-linux-gnueabihf-debug.tar.gz
11aefd547dcad2c3a8000fbf295235c3d5f2716e56b568f583510579afb3bc41  dogecoin-1.14.8-arm-linux-gnueabihf.tar.gz
3d0a2643682b3eedb5e2c2c794311492693c251de969df96b98bd09469ee582d  dogecoin-1.14.8-i686-pc-linux-gnu-debug.tar.gz
a8c6a76043f2beed00b325132916f7b3bd54434f2d0114dca17a61982c7c2a88  dogecoin-1.14.8-i686-pc-linux-gnu.tar.gz
aa587bf64972a9554cbfdcd3c049b1081e34a78de6d543ccff0da9bf83cf0a4a  dogecoin-1.14.8-x86_64-linux-gnu-debug.tar.gz
093fd4ba6d0bcce8ad9fa5c2b3a071b9996201246f348c48ddc3618397bcf900  dogecoin-1.14.8-x86_64-linux-gnu.tar.gz
ccb499769f5101535600dc8f39bb5f8257553839e06de8ebd27db3e5cec4fc4f  dogecoin-1.14.8-win-unsigned.tar.gz
b95f8d6772e5fdda5f39e112092e05bd900bbbdb5ca01030a7c217006ba0f3e1  dogecoin-1.14.8-win32-debug.zip
a47ecb82918bf9fda864039362baf99e6fa366b75bddc356e3c937aa54430ff7  dogecoin-1.14.8-win32-setup-unsigned.exe
84f0ee59233ae6a59a67c1e6ea3a247a7479c65ff23727e383545837779411f2  dogecoin-1.14.8-win32.zip
5ed71adf01f35adf84a7f388e162cb01b7f4bccb9ecd7fc1dad8826c60333ea7  dogecoin-1.14.8-win64-debug.zip
e08c7c8844fbb8bd7c2047a9fba584df5b9e0a2404df81fa935b1fc356e3272f  dogecoin-1.14.8-win64-setup-unsigned.exe
41bc51d375cafadc27cb53b6c72ab17a5cf0a3b0084008a4ff1e56685a0e0d9c  dogecoin-1.14.8-win64.zip
082a115927a974e13c9edd5c64a56a1c71b16289c3ed7590375af50439457cfc  src/dogecoin-1.14.8.tar.gz
36a5443655ee5ba915236d3a256160e3e51b9e0f677237cc272e8378032d6f96  dogecoin-win-1.14-res.yml
```
going to scrap this and start a new run with https://github.com/devrandom/gitian-builder/commit/874ba2036e4bf97ac920257c36ff29e4dea78449.

**Comment by xanimo** [comment]
ok osx is matching. will post the full batch of shasums when it's done.

**Comment by xanimo** [comment]
```
145fe5c64987df13470f63f098a927cfb1a66033528f587c886cda4b7a4746be  dogecoin-1.14.8-aarch64-linux-gnu-debug.tar.gz
7d36a74e047a4bac84bbaca9e9cc2a0fdad22cbf0d7788ca4c755b2927636942  dogecoin-1.14.8-aarch64-linux-gnu.tar.gz
d1ec2c7f0e9a987ace361749d78fa2cd5aa486f797a904d6f5bfc50d44266800  dogecoin-1.14.8-arm-linux-gnueabihf-debug.tar.gz
c3004cd651285d05fa15e00ad38158f079ced77a57adb81a8d474f7d5982b2f7  dogecoin-1.14.8-arm-linux-gnueabihf.tar.gz
a1be556915bcf4dc1d82e6b0e308859c4516f16fd82146f32ed40b8c1c0f5dca  dogecoin-1.14.8-i686-pc-linux-gnu-debug.tar.gz
6c9159fce9cb6d6f587639cdcf308edbfad16a8d7de886e01e2b6477dd67f3dd  dogecoin-1.14.8-i686-pc-linux-gnu.tar.gz
1826a71a6418c2cdbaa61c838614c8e10232553f14facb7db73e1c29bba85bea  dogecoin-1.14.8-x86_64-linux-gnu-debug.tar.gz
fb2354fc18beb708e9aff7b8f1ba62bec5c92e39f6f0ac6e4367729bd88d387c  dogecoin-1.14.8-x86_64-linux-gnu.tar.gz
e88308762d70d40bc1ec488b749221c6ed7fad859ecbcb0d596830d95712be42  dogecoin-1.14.8-win-unsigned.tar.gz
9bbda0005c8e60298177dbcf6ad4196bad9ae75da4af43b06ae89eb032f0e281  dogecoin-1.14.8-win32-debug.zip
b9feb996a622be506c20c0026270bfc507477c557ad6be0805dfe87221955ccf  dogecoin-1.14.8-win32-setup-unsigned.exe
5b29f61cdcfb5405d1d36812d6a365aef9dbb99f718bfc239689d12082fb0298  dogecoin-1.14.8-win32.zip
d044c8e9b6f8ad370acc81cad00f905e1e31f0f8a3ac483c3bb4966bd8b28270  dogecoin-1.14.8-win64-debug.zip
b3761466a402cbe1d1ce7265697eea24fcd37799a81aefc02d395ca80d3a00fb  dogecoin-1.14.8-win64-setup-unsigned.exe
580230912317d607530df3ee04e8713bafd9cd6bc03e2026926673d18446eab1  dogecoin-1.14.8-win64.zip
64fc5477b102950e47345d3acdc16f7df9fa53275075dff3bc4e4ebe32d141bf  dogecoin-1.14.8-osx-unsigned.dmg
eee529cc15d70454c9eff036e98f992b78cb8d2bd3b1d007cce8a4f9db69e767  dogecoin-1.14.8-osx-unsigned.tar.gz
171f1b234669c7e6a38522f8d78960b6652e72e36c598a96e0952de27c29fdea  dogecoin-1.14.8-osx64.tar.gz
082a115927a974e13c9edd5c64a56a1c71b16289c3ed7590375af50439457cfc  src/dogecoin-1.14.8.tar.gz
0895495f10336c7559eaa7660775bb33b497a8f36f9fdfc2886306c921f75569  dogecoin-osx-1.14-res.yml
```

**Comment by chromatic** [comment]
That's three independent gitian builds by three independent developers and all fingerprints match. We're good to go!

**Comment by chromatic** [comment]
@patricklodder given the new release process, we should update (after this release) the release documentation to describe if/how we want the final steps, such as fast-forwarding listed here.

(I think we don't need fast forwarding, but I think we do want some other branch manipulation--perhaps tagging as the final item before gitian verification?)

**Comment by patricklodder** [comment]
> I think we don't need fast forwarding, but I think we do want some other branch manipulation--perhaps tagging as the final item before gitian verification?

As you saw, I didn't make fast-forwarding to `master` a todo item here, because per #3569, `1.15.0-dev` is going to be `master` now (if/once that gets merged.)

The tag of what was the final item before gitian verification is the release tag - as that's what is provided for reproducible builds. We also cannot assume that on the main branch, gitian works just because it worked on a backported release on another branch. I'd propose to not add tags other than release tags and just keep it simple and straightforward.

Instead, I think we should just remove the [`backport to 1.14`](https://github.com/dogecoin/dogecoin/pulls?q=is%3Apr+label%3A%22backport+to+1.14%22+is%3Aclosed) tags for everything that made it into `1.14-maint`. I figure that with that, we'll just have a list of open items to backport whenever we do a minor release in the future?

**Comment by victorsk2019** [comment]
Hi,

I didn't want to interrupt with validation process, but working with gitian-build for the first time, I ran into several issues when trying to add my hash values.  Now that validation is completed, thought I'd mention these issues, for future reference:

1) encountered "permission denied" for http://172.17.0.1:3142 when building docker image `base-focal-amd64:latest` and "pull access denied" when running `gitian-build.sh` script.  I fixed these by building docker image `base-focal-amd64:latest` manually by commenting out line: `RUN echo 'Acquire::http { Proxy "http://172.17.0.1:3142"; };' > /etc/apt/apt.conf.d/50cacher` in the docker file.  Those issues were with/without `--verify`.

2) Encountered these errors when build script was finished:

```Creating package manifest
Creating build script (var/build-script)
Successfully copied 111MB to gitian-target:/home/ubuntu/build/
Running build script (log in var/build.log)
    Grabbing results from target
Grabbing cache from target
Cleaning up target
Generating report
145fe5c64987df13470f63f098a927cfb1a66033528f587c886cda4b7a4746be  dogecoin-1.14.8-aarch64-linux-gnu-debug.tar.gz
7d36a74e047a4bac84bbaca9e9cc2a0fdad22cbf0d7788ca4c755b2927636942  dogecoin-1.14.8-aarch64-linux-gnu.tar.gz
d1ec2c7f0e9a987ace361749d78fa2cd5aa486f797a904d6f5bfc50d44266800  dogecoin-1.14.8-arm-linux-gnueabihf-debug.tar.gz
c3004cd651285d05fa15e00ad38158f079ced77a57adb81a8d474f7d5982b2f7  dogecoin-1.14.8-arm-linux-gnueabihf.tar.gz
a1be556915bcf4dc1d82e6b0e308859c4516f16fd82146f32ed40b8c1c0f5dca  dogecoin-1.14.8-i686-pc-linux-gnu-debug.tar.gz
6c9159fce9cb6d6f587639cdcf308edbfad16a8d7de886e01e2b6477dd67f3dd  dogecoin-1.14.8-i686-pc-linux-gnu.tar.gz
1826a71a6418c2cdbaa61c838614c8e10232553f14facb7db73e1c29bba85bea  dogecoin-1.14.8-x86_64-linux-gnu-debug.tar.gz
fb2354fc18beb708e9aff7b8f1ba62bec5c92e39f6f0ac6e4367729bd88d387c  dogecoin-1.14.8-x86_64-linux-gnu.tar.gz
082a115927a974e13c9edd5c64a56a1c71b16289c3ed7590375af50439457cfc  src/dogecoin-1.14.8.tar.gz
d9902c20be968b5c155d6df9ea755dfb8b7ef74441e0161526581eddf01f3c4e  dogecoin-linux-1.14-res.yml
Done.
~/projects/fromgit/development
~/projects/fromgit/development/gitian-builder ~/projects/fromgit/development

Verifying v1.14.8 osx

./bin/gverify:82:in `<main>': ../gitian.sigs/1.14.8-osx does not exist (RuntimeError)

Verifying v1.14.8 win

./bin/gverify:82:in `<main>': ../gitian.sigs/1.14.8-win does not exist (RuntimeError)

Verifying v1.14.8 linux

./bin/gverify:82:in `<main>': ../gitian.sigs/1.14.8-linux does not exist (RuntimeError)
~/projects/fromgit/development
```

Running gitian-build scripts on my XPS laptop took a bit of time and so I didn't re-run it again. It was suggested to use https://github.com/devrandom/gitian-builder/commit/874ba2036e4bf97ac920257c36ff29e4dea78449).  Just for future reference, would this would have fixed the issue I encountered in 2)?  

Thanks!

**Comment by patricklodder** [comment]
> encountered "permission denied" for `http://172.17.0.1:3142/`

Did you install `apt-cacher-ng` and activate it? There's a section about that at the start of the [doc](https://github.com/dogecoin/dogecoin/blob/master/doc/gitian-building.md#common-dependencies)

**Comment by victorsk2019** [comment]
> Did you install `apt-cacher-ng` and activate it?

Nope, but I did now.  Thank you!

**Comment by chromatic** [comment]
I'm about to run these commands before I `git push`:

```
 git fetch --all
 git checkout --track upstream/1.14-maint 
 git merge --ff-only v1.14.8 
```

**Comment by patricklodder** [comment]
> I'm about to run these commands 

You'll need to override push protection and restore it after you're done.

> `git checkout --track upstream/1.14-maint `

I'd remove the `--track`, and push explicitly. That way, less chance of it going wrong.

**Comment by chromatic** [comment]
Hm, I don't see an option to override push protection in the repository settings here. How do you usually force this fast-forward?

**Comment by patricklodder** [comment]
> I don't see an option to override push protection

I've created a temporary override under `*-maint` branch security right now, so you can push.

**Comment by chromatic** [comment]
I get the same error. Do you want to try?

```
$ git push upstream 1.14-maint 
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
remote: error: GH006: Protected branch update failed for refs/heads/1.14-maint.
remote: error: Changes must be made through a pull request.
To github.com:dogecoin/dogecoin.git
 ! [remote rejected]       1.14-maint -> 1.14-maint (protected branch hook declined)
```

**Comment by patricklodder** [comment]
Done (and restored full push protection)

**Comment by patricklodder** [comment]
I'll manually sign the macOS binaries over the next hours.

**Comment by chromatic** [comment]
I can't explain it but I'll take it. Thank you! If you upload the files, I'll start making announcements. Otherwise I can get the uploads up before the end of the weekend.

**Comment by patricklodder** [comment]
> If you upload the files

Yes! I just need to do the Apple notarization

**Comment by patricklodder** [comment]
Update on notarization: I'm currently waiting on Apple Developer Support as there appears to be a need for verification of something - no idea what - before they allow the notarization process to continue (new since I did it last in March, but then, the process changes very often.)

I'll report back when I know more.

**Comment by patricklodder** [comment]
Still nothing and the process is still blocked. I doubt I'll get an answer on Sunday.

I think that for the next major release we'll need to asses this process. However, for 1.14.8, since it's a bugfix release, it would be bad if we cannot offer an upgrade path for macOS users, so I'm inclined to recommend waiting.

**Comment by chromatic** [comment]
Agreed. Let's see what we hear from Apple by tomorrow.

**Comment by xinzhongyoumeng** [comment]
Excuse me, when to upload binary ?

**Comment by patricklodder** [comment]
I've seen people complain on forums that resolving this issue sometimes may take a few weeks. Therefore I propose that:

1. We release everything including unsigned macOs binaries ASAP
2. When this is ok'd, we'll release a 1.14.8.1 with only Mac binaries - in case we need to change something to the code.

Thoughts?

**Comment by xanimo** [comment]
sgtm

**Comment by chromatic** [comment]
For 1.14.8.1 I assume we'll need to change the version number and re-run gitian. That's fine, and I'm :+1: to the plan--just confirming that implication.

**Comment by patricklodder** [comment]
>  I assume we'll need to change the version number

If there are any changes needed, then yes. If not, then no need and we can just publish the binary separately.

>  re-run gitian.

If it's just macOS then just the gitian-osx run, but yes.

**Comment by patricklodder** [comment]
Release is drafted, signed `SHA256SUMS` file is provided and binaries uploaded.

@chromatic please check, then we are a go.

**Comment by patricklodder** [comment]
accidentally pressed publish 🤣 we can still edit the text where needed.

**Comment by chromatic** [comment]
Signatures and signature of signatures verified.

**Comment by PayamShahmansour** [comment]
> There are several pending bug fixes on the 1.15.0-dev branch that benefit the network when they become available asap. Therefore, a 1.14.8 release will be beneficial. 
> 
> In scope: bug fixes, tests and build system fixes.
> Out of scope: everything else.
> 
> Work to be backported from 1.15.0-dev:
> 
> - [x] Update the CI: #3584
>     - #3467
>     - #3517
>     - #3582
> - [x] Test fixes: #3586
>     - #3572
>     - #3574
>     - #3446
> - [x] TxRequestTracker fix (separate, because backport depends on other PRs): #3592 
>     - #3577
> - [x] Other bugfixes: #3591 
>     - #3567
>     - #3568
>     - #3575
>     - #3576
> - [x] Pinned dependencies updates: #3590 
>     - #3362
>     - #3481
> - [x] Enable build on systems with newer Boost versions up to 1.85: #3588
>     - #3313
>     - #3558
>     - #3456
> - [x] Enable build on systems with newer protobuf (this requires c++17): #3589
>     - #3561
>     - #3494
>     - #3560
>     - #3490
>     - #3492
>     - #3458
> 
> Maintainer tasks:
> 
> - [x] Open 1.14.8: #3585
> - [x] Update docs
>     - #3612
>     - #3611
> - [x] Write release notes: #3609
> - [x] Fixate v1.14.8 in source #3615
> - [x] Tag v1.14.8
> - [x] Gitian builds
> - [x] Fast-fwd 1.14-maint
> - [x] Upload binaries and sha256sum files
